Remove arguments to `MlflowClient.create_run` that are now in tags

acroz opened the pull request:
This is part of work to migrate run metadata to tags by MLflow v1.0.

By removing these arguments and reading run properties from the passed tags, the client API moves closer to the v1.0 API in which things like `source_type` are not attributes of the run. For backwards compatibility, the current attributes remain set by reading them from their corresponding tags.

acroz committed:
Remove unused imports

acroz committed:
Minor test refactoring

acroz committed:
Remove unused import

acroz committed:
Tidy imports

acroz committed:
Store string representation of SourceType in tags

The changes introduced recently that store a run's `SourceType` in tags
do not convert it to a string explicilty and so they are rendered as
strings containing the underlying `int` of the enumeration.

This changes it so the human-readable string is used instead.

acroz committed:
Send meta information from projects in tags

acroz committed:
Only accept tags in MlflowClient

acroz committed:
Update tests

acroz committed:
Merge branch 'master' of github.com:mlflow/mlflow into client-create-run-only-tags

acroz committed:
Fix tests

acroz commented:
Tagging @aarondav and @mparkhe for context.

acroz committed:
Merge branch 'master' into client-create-run-only-tags

aarondav commented on mlflow/tracking/client.py:
```diff
@@ -52,18 +52,35 @@ def create_run(self, experiment_id, user_id=None, run_name=None, source_type=Non
+        # Extract run attributes from tags
+        # This logic is temporary; by the 1.0 release, this information will only be stored in tags
+        # and will not be available as attributes of the run
+        parent_run_id = tags.get(MLFLOW_PARENT_RUN_ID)
```
`parent_run_id` we actually intentionally added as an extra parameter to the create_run call. The idea here was that the notion of linking two runs may be fundamental to the creation of those runs, and may imply other properties (for example, we might update metadata on the parent run).

In practice, we haven't yet implemented any such extra linking from an API perspective, so it may be OK to degrade this to just be a tag instead of a property of the create call.

acroz replied:
The parent run ID does seem like something that we might want to have as a strongly integrated attribute on runs, but while it is exposed as a 'top level' attribute when creating runs, it seems to only be accessible as a tag when getting a run.

I would suggest that it should either be 'just' a tag and set as in this PR, or it should be promoted to a full attribute like `start_time` that appears in the `RunInfo` object itself.

Since it's currently stored as a tag, and adding it to the `RunInfo` object will require changing the protobuf files etc., I suggest keeping it as a tag for the time being and promoting it to be an actual attribute later if we think it's valuable.

acroz replied:
(FYI my last commit is not meant to assume the above decision - just don't want to forget to update the docs in the case my proposal is accepted)

aarondav replied:
Alright, sounds reasonable to me.

acroz committed:
Add comment on deprecation of parent_run_id attribute

As discussed with @aarondav in #978, this might not get deprecated. I'm
adding it to this branch though as I'll probably forget if we decide to
make the parent run ID just be stored as a tag.
